osu.Game/Screens/Menu/ButtonSystem.cs

```diff
@@ -81,10 +81,66 @@ public void SetOsuLogo(OsuLogo logo)
         private readonly ButtonArea buttonArea;
 
         private readonly MainMenuButton backButton;
+        private readonly MainMenuButton settingsButton;
+
+        // public for tests
+        [CanBeNull]
+        public MainMenuButton PlayButton { get; private set; }
 
         private readonly List<MainMenuButton> buttonsTopLevel = new List<MainMenuButton>();
         private readonly List<MainMenuButton> buttonsPlay = new List<MainMenuButton>();
 
+        [CanBeNull]
+        public List<MainMenuButton> CurrentButtonsList => State switch
+        {
+            ButtonSystemState.TopLevel => buttonsTopLevel,
+            ButtonSystemState.Play => buttonsPlay,
+            _ => null
+        };
+
+        private int selectionIndex = -1;
```

Comment:
Would really like this to be `int?` and for `null` to denote the "no selection" case rather than use C-style special values.